src/interface/sparse_pack.hpp

```diff
@@ -114,6 +126,19 @@ struct base_t {
   static int size() { return multiply<NCOMP...>::value; }
 
   const int idx;
+
+ private:
+  template <int Head, int... Tail, typename... TailArgs,
+            REQUIRES(all_implement<integral(TailArgs...)>::value)>
+  KOKKOS_INLINE_FUNCTION static auto GetIndex_(int first, TailArgs... rest) {
+    if constexpr (sizeof...(Tail) == 0)
+      return first;
+    else if constexpr (sizeof...(TailArgs) == 0)
+      return first;
+    else
+      return (multiply<Tail...>::value * first +
+              GetIndex_<Tail...>(std::forward<TailArgs>(rest)...));
+  }
```

Comment:
FWIW, I think this could be done with a fold expression as 
```c++
template <class... Args>
KOKKOS_INLINE_FUNCTION static auto GetIndex_(Args... args) {
  int idx = 0; 
  ([&]{idx *= NCOMP; idx += args}(), ...);
  return idx;
}
```

Reply:
@mari2895 if you want to use this, you should first double check that it still works (i.e., that tests pass). You also still want to require args are integral type.

Reply:
@Yurlungur @lroberts36 ok I am going try out this change and check if it works.